pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -168,6 +181,7 @@ func vmStatsCollectorCallback() []operatormetrics.CollectorResult {
 	}
 
 	var results []operatormetrics.CollectorResult
+	results = append(results, CollectDiskAllocatedSize(vms)...)
 	results = append(results, CollectVMsInfo(vms)...)
 	results = append(results, CollectResourceRequestsAndLimits(vms)...)
 	results = append(results, reportVmsStats(vms)...)
```

Comment:
Consider making these unexported

Reply:
at the time I started exporting these functions so that we could unlink the unit test code from the product code

Reply:
absolutely right! fixed

Reply:
It just depends on if we want to perform black-box or white-box testsing.

Reply:
I think that if we decide to move the tests then we can change it again to be exported. is it ok @machadovilaca?

Reply:
ok so @machadovilaca mentioned unit test will fail if changing it, so I fixed it back to be exported.
we can discusses it here and if we decide to change we can do it in a separate pr

Reply:
Ok but unit tests don't fail, if proper renamed:
```
Will run 72 of 72 specs
••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••;5;10m••••••••••••

Ran 72 of 72 Specs in 0.002 seconds
SUCCESS! -- 72 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestVirtController (0.00s)
PASS

Process finished with the exit code 0
```

Reply:
> It just depends on if we want to perform black-box or white-box testing.

what would be the disadvantage of this approach? these are called directly in the Prometheus callback, we just skip the `vmInformer.GetIndexer().List()` in the tests basically
and this way we can test each specific collect individually

what would be your suggestion? unexporting these functions and test the callback as a whole?

Reply:
To me, we can just perform white-box testing, which means unexporting the functions and test'em as we are doing right now in the test file.
We are already doing it in most of the controller tests.
The issue with unexported functions will appear only if you want to switch the test package to `*_test` (black-box testing). But it's up to you, I will not force anyone to use white-box testing. There are advantages and disadvantages to both strategies.

Reply:
yes, I was looking into switching to `*_test`, I thought it was the direction we were trying to go, saw some PRs with those kind of changes

Reply:
I think we should open a wider discussion to define what is our direction :)